app/services/sync_provider_from_find.rb

```diff
@@ -10,22 +14,30 @@ def self.call(provider_code:)
       .find(provider_code)
       .first
 
-    provider = create_provider(find_provider)
+    update_provider(provider, find_provider.provider_name)
 
     find_provider.courses.each do |find_course|
-      create_course(find_course, provider)
+      create_or_update_course(find_course, provider)
     end
   end
 
-  def self.create_provider(find_provider)
-    provider = Provider.find_or_create_by(code: find_provider.provider_code)
-    provider.name = find_provider.provider_name
-    provider.save!
+  def self.create_or_update_provider(provider_name, provider_code, sync_courses)
+    provider = Provider.find_or_create_by(code: provider_code)
+    if sync_courses
+      provider.sync_courses = sync_courses
+      provider.save!
+    end
+    update_provider(provider, provider_name) if provider_name.present?
 
     provider
   end
 
-  def self.create_course(find_course, provider)
+  def self.update_provider(provider, provider_name)
```

Comment:
This method feels quite specialised for its name

Reply:
Looks like it should be `update_provider_name`. Changing...